lib/blockscore/base.rb

```diff
@@ -53,22 +52,22 @@ def self.resource
       @resource ||= Util.to_underscore(to_s.split('::').last)
     end
 
-    def self.api_url
-      'https://api.blockscore.com/'
-    end
-
     def self.endpoint
       fail NotImplementedError, ABSTRACT_WARNING if equal?(Base)
 
-      "#{api_url}#{Util.to_plural(resource)}"
+      API_URL + Util.to_plural(resource)
     end
 
     def persisted?
-      !id.nil? && !attributes[:deleted]
+      !id.nil? && !attributes.fetch(:deleted, false)
     end
 
     protected
```

Comment:
I will see how `protected`plays into this code. It is rarely used. One of my concerns which I have not addressed is the overall landscape of methods that are publicly accessible. It feels like it is too broad.